pandas/tests/sparse/test_combine_concat.py

```diff
@@ -79,49 +113,67 @@ def test_concat_different_kind(self):
         val2 = np.array([3, np.nan, 4, 0, 0])
 
         sparse1 = pd.SparseSeries(val1, name='x', kind='integer')
-        sparse2 = pd.SparseSeries(val2, name='y', kind='block', fill_value=0)
+        sparse2 = pd.SparseSeries(val2, name='y', kind='block')
 
         res = pd.concat([sparse1, sparse2])
         exp = pd.concat([pd.Series(val1), pd.Series(val2)])
-        exp = pd.SparseSeries(exp, kind='integer')
+        exp = pd.SparseSeries(exp, kind=sparse1.kind)
         tm.assert_sp_series_equal(res, exp)
 
         res = pd.concat([sparse2, sparse1])
         exp = pd.concat([pd.Series(val2), pd.Series(val1)])
-        exp = pd.SparseSeries(exp, kind='block', fill_value=0)
-        tm.assert_sp_series_equal(res, exp)
-
-    def test_concat_sparse_dense(self):
+        exp = pd.SparseSeries(exp, kind=sparse2.kind)
+        tm.assert_sp_series_equal(res, exp, consolidate_block_indices=True)
+
+    @pytest.mark.parametrize('kind', [
+        'integer',
+        'block',
+    ])
+    def test_concat_sparse_dense(self, kind):
         # use first input's fill_value
         val1 = np.array([1, 2, np.nan, np.nan, 0, np.nan])
         val2 = np.array([3, np.nan, 4, 0, 0])
 
-        for kind in ['integer', 'block']:
-            sparse = pd.SparseSeries(val1, name='x', kind=kind)
-            dense = pd.Series(val2, name='y')
-
-            res = pd.concat([sparse, dense])
-            exp = pd.concat([pd.Series(val1), dense])
-            exp = pd.SparseSeries(exp, kind=kind)
-            tm.assert_sp_series_equal(res, exp)
-
-            res = pd.concat([dense, sparse, dense])
-            exp = pd.concat([dense, pd.Series(val1), dense])
-            exp = pd.SparseSeries(exp, kind=kind)
-            tm.assert_sp_series_equal(res, exp)
-
-            sparse = pd.SparseSeries(val1, name='x', kind=kind, fill_value=0)
-            dense = pd.Series(val2, name='y')
+        sparse = pd.SparseSeries(val1, name='x', kind=kind)
+        dense = pd.Series(val2, name='y')
 
-            res = pd.concat([sparse, dense])
-            exp = pd.concat([pd.Series(val1), dense])
-            exp = pd.SparseSeries(exp, kind=kind, fill_value=0)
-            tm.assert_sp_series_equal(res, exp)
+        res = pd.concat([sparse, dense])
+        exp = pd.concat([pd.Series(val1), dense])
+        exp = pd.SparseSeries(exp, kind=kind)
```

Comment:
there is twice `exp` here

Reply:
I think deliberate, to ensure we get the right dtype & kind.

Reply:
Ah, I missed that `exp` was passed to the second line ..